tests/taso_termination.rs

```diff
@@ -0,0 +1,34 @@
+#[cfg(feature = "portmatching")]
+mod require_portmatching {
+    use hugr::Hugr;
+    use rstest::{fixture, rstest};
+    use tket2::{
+        json::TKETDecode,
+        optimiser::{DefaultTasoOptimiser, TasoOptimiser},
+        Circuit,
+    };
+    use tket_json_rs::circuit_json::SerialCircuit;
+
+    #[fixture]
+    fn nam_4_2() -> DefaultTasoOptimiser {
+        // TasoOptimiser::default_with_rewriter_binary("test_files/nam_4_2.rwr").unwrap()
+        TasoOptimiser::default_with_eccs_json_file("test_files/Nam_4_2_complete_ECC_set.json")
+            .unwrap()
+    }
+
+    #[fixture]
+    fn simple_circ() -> Hugr {
+        let json = r#"
+        {"bits": [], "commands": [{"args": [["q", [0]], ["q", [2]]], "op": {"type": "CX"}}, {"args": [["q", [0]]], "op": {"params": ["0.1"], "type": "Rz"}}, {"args": [["q", [0]], ["q", [1]]], "op": {"type": "CX"}}, {"args": [["q", [1]]], "op": {"type": "H"}}, {"args": [["q", [1]]], "op": {"params": ["0.2"], "type": "Rz"}}, {"args": [["q", [1]]], "op": {"type": "H"}}, {"args": [["q", [0]], ["q", [1]]], "op": {"type": "CX"}}, {"args": [["q", [0]], ["q", [2]]], "op": {"type": "CX"}}, {"args": [["q", [0]]], "op": {"params": ["-0.1"], "type": "Rz"}}], "created_qubits": [], "discarded_qubits": [], "implicit_permutation": [[["q", [0]], ["q", [0]]], [["q", [1]], ["q", [1]]], [["q", [2]], ["q", [2]]]], "phase": "0.0", "qubits": [["q", [0]], ["q", [1]], ["q", [2]]]}
+    "#;
+        let ser: SerialCircuit = serde_json::from_str(json).unwrap();
+        ser.decode().unwrap()
+    }
+
+    #[rstest]
+    #[ignore = "Takes 800ms"]
```

Comment:
800ms is not a reasonable test time.

With the latest updates, this test now takes 340ms in. This is slower than running all the other 52 tests in this crate combined (230ms).

I say we merge this now, but leave the commented-out ignore as a marker in case test times ever become a problem.